LeaderboardBackend.Test/TestApi/TestApiFactory.cs

```diff
@@ -85,6 +69,38 @@ public TestApiClient CreateTestApiClient()
         return new TestApiClient(client);
     }
 
+    public void InitializeDatabase()
+    {
+        using IServiceScope scope = Services.CreateScope();
+        ApplicationContext dbContext = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        InitializeDatabase(dbContext);
+    }
+
+    private static void InitializeDatabase(ApplicationContext dbContext)
+    {
+        switch (TestConfig.DatabaseBackend)
+        {
+            case DatabaseBackend.TestContainer:
+                if (!PostgresDatabaseFixture.HasCreatedTemplate)
+                {
+                    dbContext.MigrateDatabase();
+                    Seed(dbContext);
+                    PostgresDatabaseFixture.CreateTemplateFromCurrentDb();
+                }
+
+                break;
+            case DatabaseBackend.InMemory:
```

Comment:
Are we not dropping support for in-memory DBs due to them not supporting the constraints we need? I seem to remember some discussion around that.

Reply:
We haven't formally decided, though I think we should. But it's not the purpose of the PR.